### PR TITLE
[incubator-kie-issues-1204] Add built in support for org.w3c.dom.Document

### PR DESCRIPTION
**Pull request by elguardian**
Closes: https://github.com/apache/incubator-kie-issues/issues/1204

### CODE REVIEW

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/proto/ProtoGenerator.java [hunk 1]
```diff
@@ -44,76 +36,6 @@ public interface ProtoGenerator {
 
     Collection<GeneratedFile> generateProtoFiles();
 
-    default String applicabilityByType(String type) {
```
Comment (martinweiler) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/proto/ProtoGenerator.java:47]: This requires another change:
```
[ERROR] quarkus/extensions/kogito-quarkus-workflow-extension-common/kogito-quarkus-workflow-common-deployment/src/main/java/org/kie/kogito/quarkus/workflow/deployment/JandexProtoGenerator.java:[160,54] cannot find symbol
[ERROR]   symbol:   method applicabilityByType(java.lang.String)
[ERROR]   location: class org.kie.kogito.quarkus.workflow.deployment.JandexProtoGenerator
```

**Comment by martinweiler** [comment]
Still seeing 
```
[ERROR] Caused by: java.lang.IllegalArgumentException: Custom type org.w3c.dom.Document is no supported by Kogito persistence, please consider using a class that extends java.io.Serializable and contains a no arg constructor
[ERROR] 	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.protoType(AbstractProtoGenerator.java:259)
```
in a kogito app with a process containing a `org.w3c.dom.Document` process variable.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/datatype/impl/coverter/TypeConverterRegistry.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 import java.util.Map;
 import java.util.function.Function;
 
+import javax.swing.text.Document;
```
Comment (martinweiler) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/datatype/impl/coverter/TypeConverterRegistry.java:26]: Is this intentional? Shouldn't this be `org.w3c.dom.Document`?

Comment (elguardian) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/datatype/impl/coverter/TypeConverterRegistry.java:26]: nope

Comment (elguardian) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/datatype/impl/coverter/TypeConverterRegistry.java]: interestingly when I did try to set an object property with the dom the rest endpoint was converting it :S

**Review by martinweiler — APPROVED** [review]

**Review by pefernan — APPROVED** [review]

**Pull request merged**